### PR TITLE
feat: support single quote flags

### LINKED ISSUE #258: Labels in flags wont work due to below logic
https://github.com/google-github-actions/deploy-cloudrun/blob/92e1db9e55efcb6ddc60ef91a3e948938a97b94e/src/deploy-cloudrun.ts#L239

### LINKED ISSUE #268: single quotes causes `=` trim
Hey there,

Is there any reason as to why we're not accepting single quotes here? If formatting all flag parameters with double quotes as starting delimiters and single quotes as value delimiters I get all my `=`s escaped. If this is as intended `// Split on space or "=" if not in quotes` -> `// Split on space or "=" if not in double quotes`

https://github.com/google-github-actions/deploy-cloudrun/blob/07a7b6c4efab20a52aa35710084ce16bb70a7d3b/src/deploy-cloudrun.ts#L239

### PR DESCRIPTION
**Pull request by sethvargo**
Fixes #268 
Fixes #258

### CODE REVIEW

File: tests/unit/deploy-cloudrun.test.ts [hunk 1]
```diff
@@ -173,20 +173,40 @@ describe('#deploy-cloudrun', function () {
...
+    it('parses flags using equals and single quotes', async function () {
+      const input = `--entry-point='node index.js'`;
+      const results = parseFlags(input);
+      expect(results).to.eql(['--entry-point', "'node index.js'"]);
+    });
+    it('parses flags mixing double and single quotes correctly', async function () {
+      const input = `--entry-point 'node index.js' --memory "2 Gi"`;
+      const results = parseFlags(input);
+      expect(results).to.eql(['--entry-point', "'node index.js'", '--memory', '"2 Gi"']);
+    });
```
Comment (bharathkkb) [inline tests/unit/deploy-cloudrun.test.ts:210]: Looks like a good candidate for a table test

**Review by bharathkkb — COMMENTED** [review]
LGTM minor suggestion

File: tests/unit/deploy-cloudrun.test.ts (hunk 1 above)
Comment (sethvargo) [reply tests/unit/deploy-cloudrun.test.ts:210]: Yea, I planned to do this in a follow-up so it was a unique body of work. There's another test in this file that could benefit from being table-driven, but I wanted to minimize the diff.

Comment (sethvargo) [reply tests/unit/deploy-cloudrun.test.ts:210]: https://github.com/google-github-actions/deploy-cloudrun/pull/290

Comment (sethvargo) [reply tests/unit/deploy-cloudrun.test.ts]: Actually nvm - the diff isn't so bad. Added here.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**